### DISCUSSION TITLE
React.js deployment error: blocked by CORS policy

### DISCUSSION THREAD

**Discussion by ngcxy** [vercel/community#3279] [Help] [answered]
### Summary

I'm working on deploying a react.js frontend to Vercel. It's connected to a Django backend, which is already deployed on Vercel successfully. 

I have read the article: [how-to-enable-cors](https://vercel.com/guides/how-to-enable-cors) I followed the instruction to create the versel.json. This is its content: 
```
{
  "headers": [
    {
      "source": "/(.*)",
      "headers": [
        { "key": "Access-Control-Allow-Credentials", "value": "true" },
        { "key": "Access-Control-Allow-Origin", "value": "*" },
        { "key": "Access-Control-Allow-Methods", "value": "GET,OPTIONS,PATCH,DELETE,POST,PUT" },
        { "key": "Access-Control-Allow-Headers", "value": "X-CSRF-Token, X-Requested-With, Accept, Accept-Version, Content-Length, Content-MD5, Content-Type, Date, X-Api-Version" }
      ]
    }
  ]
}
```

I'm unfamiliar with next.js so I just completed this step in vercel.json.

I'm a beginner in deploying website. I appreciate any help and discussion that you can share.

### Example

my website on vercel: https://my-blog-app-silk.vercel.app/

### Steps to Reproduce

my github repository for react: [myBlog-frontend](https://github.com/ngcxy/myBlog-frontend)

the error in the console:
`Access to XMLHttpRequest at 'https://my-blog-backend-delta.vercel.app/api/token/' from origin 'https://my-blog-app-silk.vercel.app' has been blocked by CORS policy: Response to preflight request doesn't pass access control check: No 'Access-Control-Allow-Origin' header is present on the requested resource.`

`POST https://my-blog-backend-delta.vercel.app/api/token/ net::ERR_FAILED`

**Comment by ngcxy** [comment, marked as answer]
I managed to resolve the issue on my own. It turns out the problem was with the server. I add the header config in vercel.js, and also change the 'CORS_ALLOWED_ORIGINS' variable in the setting, and it works.

**Reply by irohitb** [reply]
> CORS_ALLOWED_ORIGINS' variable in the setting. 

Where in setting can I find it?

**Reply by ngcxy** [reply]
If your backend is same as me with Django, then it should be the settings.py file which is in the folder of the main app. I'm not sure it works on other backend frameworks. Hope this can help you a little bit.

**Reply by megha8191** [reply]
what if its just frontend project with no backend?
